packages/components/pages/f-checkout/src/components/Notes.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+    <div v-if="isSplitNotesEnabled">
+        <accordion
+            :id="noteTypeCourierOrOrder"
+            :title="$t(`userNote.${noteTypeCourierOrOrder}.${serviceType}.title`)">
+            <span :class="$style['c-checkout-accordion-help']">{{ $t(`userNote.${noteTypeCourierOrOrder}.${serviceType}.text`) }}</span>
+            <form-field
+                :placeholder="$t(`userNote.${noteTypeCourierOrOrder}.${serviceType}.placeholder`)"
+                :value="noteValue"
+                v-bind="inputStyles"
+                :name="`${noteTypeCourierOrOrder}-note`"
+                @input="updateUserNotes({ note: $event, type: noteTypeCourierOrOrder })" />
+        </accordion>
+        <accordion
+            v-if="shouldShowKitchenNotes"
+            id="kitchen"
+            :title="$t(`userNote.kitchen.${serviceType}.title`)">
+            <span :class="$style['c-checkout-accordion-help']">{{ $t(`userNote.kitchen.${serviceType}.text`) }}</span>
+            <form-field
+                :placeholder="$t(`userNote.kitchen.${serviceType}.placeholder`)"
+                :value="kitchenNoteValue"
+                v-bind="inputStyles"
+                name="kitchen-note"
+                @input="updateUserNotes({ note: $event, type: 'kitchen' })" />
+        </accordion>
+    </div>
+    <form-field
+        v-else-if="!isSplitNotesEnabled"
+        :label-text="$t(`userNote.order.delivery.title`)"
+        :placeholder="$t(`userNote.order.${serviceType}.placeholder`)"
+        :value="noteValue"
+        v-bind="inputStyles"
+        :name="'order-note'"
+        @input="updateUserNotes({ note: $event, type: 'order' })" />
+</template>
+
+<script>
+import { mapActions, mapState } from 'vuex';
+import FormField from '@justeat/f-form-field';
+import Accordion from './Accordion.vue';
+import '@justeat/f-form-field/dist/f-form-field.css';
+import {
+    CHECKOUT_NOTE_TYPE_ORDER,
+    CHECKOUT_NOTE_TYPE_COURIER,
+    VUEX_CHECKOUT_MODULE
+} from '../constants';
+import loggerMixin from '../mixins/logger.mixin';
+
+export default {
+    components: {
+        Accordion,
+        FormField
+    },
+
+    mixins: [
+        loggerMixin
+    ],
+
+    props: {
+        isSplitNotesEnabled: {
+            type: Boolean,
+            default: false
+        }
+    },
+
+    computed: {
+        ...mapState(VUEX_CHECKOUT_MODULE, [
+            'notesConfiguration',
+            'serviceType',
+            'notes'
+        ]),
+
+        shouldShowKitchenNotes () {
```

Comment:
This could be a getter in the store?

Reply:
And same for lots of the below computed properties, which seem to only depend on values from the store's state

Reply:
Yep, I've decided to take this approach with a few of the things you've suggested.

Reply:
Now updated with no computed properties in the notes component!